metrics_sdk/lib/opentelemetry/sdk/metrics/export/periodic_metric_reader.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Export
+        # PeriodicMetricReader provides a minimal example implementation.
+        class PeriodicMetricReader < MetricReader
+          def initialize(interval_millis: 60, timout_millis: 30, exporter: nil)
+            super()
+
+            @interval_millis = interval_millis
+            @timout_millis = timout_millis
+            @exporter = exporter
+            @thread   = nil
+            @continue = false
+
+            start
+          end
+
+          def start
+            @continue = true
+            if @exporter.nil?
+              OpenTelemetry.logger.warn 'Missing exporter in PeriodicMetricReader.'
+            elsif @thread&.alive?
+              OpenTelemetry.logger.warn 'PeriodicMetricReader is still running. Please close it if it needs to restart.'
+            else
+              @thread = Thread.new do
+                while @continue
+                  sleep(@interval_millis)
+                  begin
+                    Timeout.timeout(@timout_millis) { @exporter.export(collect) }
+                  rescue Timeout::Error => e
+                    OpenTelemetry.handle_error(exception: e, message: 'PeriodicMetricReader timeout.')
+                  end
+                end
+              end
+            end
+          end
+
+          def close
+            @continue = false   # force termination in next iteration
+            @thread.join(5)     # wait 5 seconds for collecting and exporting
```

Comment:
Why 5 seconds? Should it be configurable?

Reply:
I think I take reference on what how Java handle the thread waiting ([here](https://github.com/open-telemetry/opentelemetry-java/blob/13ea3346a9f1062fd3cc3963602911aca546e1d1/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/PeriodicMetricReader.java#L96)). I don't see spec mention about the thread waiting time but it seems like it should be configurable.

Reply:
Yes, I like using `@interval_millis` in join as well. Updated.